src/js/stores/search.js

```diff
@@ -0,0 +1,212 @@
+
+        /**
+         * Search types
+         *
+         * All handleExecute functions must return a boolean indicating if 
+         * the search item was successfully handled
+         * 
+         * @type {{ getOptions: function, filters: Immutable.List.<string>, handleExecute: function }}
+         */
+        _registeredSearchTypes: {},
+
+        initialize: function () {
+            this.bindActions(
+                events.RESET, this._handleReset,
+                events.search.REGISTER_SEARCH_PROVIDER, this._registerSearchType
+            );
+            this._handleReset();
+        },
+
+        /**
+         * Reset or initialize store state.
+         *
+         * @private
+         */
+        _handleReset: function () {
+            this._registeredSearches = {};
+            this._registeredSearchTypes = {};
+        },
+
+        /**
+         * Update the search items for the specified search module
+         *
+         * @private
+         * @param {string} id The search module ID to update
+         */
+        _update: function (id) {
+            this._updateSearchItems(id);
+        },
+
+        /**
+         * Get the search items for the specified search module
+         *
+         * @param {string} id The search module ID 
+         * @return {Immutable.Iterable.<object>}
+         */
+        getSearchItems: function (id) {
+            return this._registeredSearches[id].searchItems;
+        },
+
+        /**
+         * Get the filters for the specified search module
+         *
+         * @param {string} id The search module ID 
+         * @return {Array.<Array.<string>>}
+         */
+        getFilters: function (id) {
+            return this._registeredSearches[id].searchFilters;
+        },
+
+        /**
+         * Get the headers for the specified search module
+         *
+         * @param {string} id The search module ID
+         * @return {Array.<string>}
+         */
+        getHeaders: function (id) {
+            return this._registeredSearches[id].searchHeaders;
+        },
+
+        /**
+         * Perform action for item
+         *
+         * @param {string} id Search module to use
+         * @param {string} itemID Search item to handle
+         */
+        handleExecute: function (id, itemID) {
+            return this._defaultHandleExecute(id, itemID);
+        },
+
+        /**
+         * Add search module and find initial items, headers, filters
+         *
+         * @param {string} id The search module ID
+         * @param {Array.<string>} types What to search for
+         */
+        registerSearch: function (id, types) {
+            this._registeredSearches[id] = { "searchHeaders": types };
+            this._updateSearchItems(id);
+        },
+
+        /**
+         * Add search type
+         *
+         * @param {object} payload
+         */
+        _registerSearchType: function (payload) {
+            this._registeredSearchTypes[payload.type] = {
```

Comment:
I think you can do a lot more weight lifting in this function here, per my comments above. 

Also, please expand on what's expected in the payload, similar to this:

```
* @param {object} descriptor
     * @param {number} descriptor.layerKind
     * @param {boolean?} descriptor.artboardEnabled If set, will parse artboard value
     * @param {object?} descriptor.artboard Contains the artboard bounds descriptor
     * @param {object?} descriptor.pathBounds If available, will be parsed as shape layer
     * @param {object?} descriptor.boundsNoEffects Bounds object available for all layers
```